extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/MinMax.java

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.annotations.CopyableStyle;
+import io.deephaven.util.compare.ObjectComparisons;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Immutable;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.annotation.Nullable;
+
+@Immutable
+@CopyableStyle
+@VisibleForTesting
+public abstract class MinMax<T extends Comparable<T>> {
```

Comment:
We need to be very careful in how this relates to DH filter comparisions / sort ordering, especially in regards to floating points. 

The parquet definition of min / max _may_ need to be modified to account for how Deephaven expects to handle things. In particular, DoubleRangeFilter uses `io.deephaven.util.compare.DoubleComparisons`, you'll need to read up on that documentation to make sure you understand it all.

https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L1085-L1092

NULL_DOUBLE will also need to be accounted for, in the case where parquet is not using null presence bit... Char is also an interesting case b/c we use Character.MAX_VALUE for NULL_CHAR.